### PR TITLE
patterns: don't recurse with !/--exclude for pf:, fixes #2509

### LINKED ISSUE #2509: ! pf:/some/dir Does not exclude directory
I have a folder `~/borg-test` containing a file `f0` and directories `d1` and `d2` containing files `f1` and `f2` respectively. When I created an archive of it using the following patterns file:
```
R /home/me/borg-test
! pf:/home/me/borg-test/d1
```
it included all files and directories, and did not exclude `d1`.
Using `-` instead of `!` didn't exclude it either.

This is using the standalone binary borg-linux64 of version 1.1.0b5

### PR DESCRIPTION
**Pull request by edgimar**

### CODE REVIEW

File: src/borg/patterns.py [hunk 1]
```diff
@@ -150,9 +150,9 @@ def match(self, path):
 
         if value is not non_existent:
             # we have a full path match!
-            # TODO: get from pattern; don't hard-code
-            self.recurse_dir = True
-            return value
+            self.recurse_dir = (value is IECommand.ExcludeNoRecurse)
```
Comment (enkore) [inline src/borg/patterns.py]: This looks like it's inverted, setting recurse_dir to True when value says Exclude*NoRecurse*.

Comment (ThomasWaldmann) [inline src/borg/patterns.py]: seen at another place:
```
recurse_dir = cmd not in [IECommand.ExcludeNoRecurse]
```

Comment (edgimar) [inline src/borg/patterns.py]: sorry, yes it was inverted - fixed.

File: src/borg/patterns.py [hunk 2]
```diff
@@ -150,9 +150,9 @@ def match(self, path):
 
         if value is not non_existent:
             # we have a full path match!
-            # TODO: get from pattern; don't hard-code
-            self.recurse_dir = True
-            return value
+            self.recurse_dir = (value is not IECommand.ExcludeNoRecurse)
+
```
Comment (ThomasWaldmann) [inline src/borg/patterns.py]: hmm, an empty line here is not really needed/helpful, is it?

also, can we do the same check at both places in the same way? (see my previous comment)

**Comment by ThomasWaldmann** [comment]
@edgimar I'ld like to release b6 this weekend, can you finish the PR?

**Comment by edgimar** [comment]
@ThomasWaldmann should be OK now.

File: src/borg/patterns.py [hunk 3]
```diff
@@ -325,6 +324,11 @@ class IECommand(Enum):
     ExcludeNoRecurse = 5
 
 
+def command_recurses_dir(cmd):
+    # TODO?: raise error or return None if *cmd* is RootPath or PatternStyle
```
Comment (ThomasWaldmann) [inline src/borg/patterns.py]: guess a ValueError / TypeError might be appropriate.

**Pull request merged**